File: frontend/entities/PaymentMethod.ts
import ParticipantEvent from "./ParticipantEvent";

export default class PaymentMethod {
    id: string;
    methodName: string;
    methodDescription: string;
    active: boolean;
    additionalInformation: string;
    participantEvents: ParticipantEvent[];

    constructor(
        id: string,
        methodName: string,
        methodDescription: string,
        active: boolean,
        additionalInformation: string,
        participantEvents: ParticipantEvent[]
    ) {
        this.id = id;
        this.methodName = methodName;
        this.methodDescription = methodDescription;
        this.active = active;
        this.additionalInformation = additionalInformation;
        this.participantEvents = participantEvents;
    }
}

File: backend/App.DTO/v1_0/PaymentMethod.cs
using System.ComponentModel.DataAnnotations;
using Base.DTO;


namespace App.DTO.v1_0;

public class PaymentMethod: BaseEntity
{
    [MaxLength(256)]
    public string MethodName { get; set; } = default!;
    
    [MaxLength(256)]
    public string MethodDescription { get; set; } = default!;
    public bool Active { get; set; }
    
    public ICollection<ParticipantEvent>? ParticipantEvents { get; set; }
}